ui/src/app/views/cohort-edit/component.spec.ts

```diff
@@ -0,0 +1,267 @@
+              return true;
+            }
+          }));
+          workspaceMatch.cohorts.push(newCohort);
+          observer.complete();
+        }
+        observer.error(new Error('Error creating a cohort. No workspace '
+          + wsNamespace + ', ' + wsId + ' in cohort service stub.'));
+      }, 0);
+    });
+    return observable;
+  }
+
+  public getCohortsInWorkspace(wsNamespace: string,
+                               wsId: string): Observable<CohortListResponse> {
+    const observable = new Observable(observer => {
+      setTimeout(() => {
+        const workspaceMatch = this.getMatchingWorkspace(wsNamespace, wsId);
+        if (workspaceMatch != null) {
+          observer.next({items: workspaceMatch.cohorts});
+          observer.complete();
+        }
+        observer.error('No workspace ' + wsNamespace + ', ' + wsId + ' found.');
+      }, 0);
+    });
+    return observable;
+  }
+}
+
+const activatedRouteStub  = {
+  snapshot: {
+    url: [
+      {path: 'workspace'},
+      {path: WorkspaceComponent.DEFAULT_WORKSPACE_NS},
+      {path: WorkspaceComponent.DEFAULT_WORKSPACE_ID},
+      {path: 'cohorts'},
+      {path: 'create'}
+    ]
+  }
+};
+
+describe('CohortEditComponent', () => {
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        RouterTestingModule,
+        FormsModule
+      ],
+      declarations: [
+        CohortEditComponent
+      ],
+      providers: [
+        { provide: CohortsService, useValue: new CohortsServiceStub() },
+        { provide: ActivatedRoute, useValue: activatedRouteStub }
+      ] }).compileComponents();
+  }));
+
+
+  it('displays blank input fields when creating a new cohort', async(() => {
+    const fixture = TestBed.createComponent(CohortEditComponent);
+    fixture.detectChanges();
+    const nameField = fixture.debugElement.query(By.css('#name'));
+    const descriptionField = fixture.debugElement.query(By.css('#description'));
+    fixture.detectChanges();
+    setTimeout(function(){
+      expect(nameField.nativeNode.value).toMatch('');
+      expect(descriptionField.nativeNode.value).toMatch('');
+    }, 0);
+  }));
+
+
+  it('fetches and displays an existing cohort in the edit pane',
+  fakeAsync(() => {
+    const fixture = TestBed.createComponent(CohortEditComponent);
+    const route = fixture.debugElement.injector.get(ActivatedRoute).snapshot;
+    route.url[4].path = '1';
+    route.url.push(new UrlSegment('edit', {}));
+
+    updateAndTick(fixture);
+    const nameField = fixture.debugElement.query(By.css('#name'));
+    const descriptionField = fixture.debugElement.query(By.css('#description'));
+
+    updateAndTick(fixture);
+
+    expect(nameField.nativeElement.value).toMatch('sample name');
+    expect(descriptionField.nativeElement.value).toMatch('sample description');
+  }));
+
+  it('adds a new cohort with given name and description', fakeAsync(() => {
+    const fixture = TestBed.createComponent(CohortEditComponent);
```

Comment:
I'll look more into the `beforeEach` function, but I am currently leaning a bit more towards a getContext function and a struct, because I think `beforeEach` has some side effects/is very specifically designed for compilation and not variable assignment. (Its scope appears not to carry over.)